### PR TITLE
ISSUE-242 : Enabling chapters button only for multi chapter book

### LINKED ISSUE #242: Books with a single chapter should be handled as books without chapters
Easy enough to detect and fixing shouldn't be that hard after #231

### PR DESCRIPTION
**Pull request by mmurali88**
Proposed changes

Books with a single chapter should be handled as books without chapters. 

Types of changes

- [x]  Feature modification (non-breaking)

Checklist

- [x] I have read the CONTRIBUTING doc
- [x]  Linter check and compiled locally before pushing 
- [x] Tested multi chapter / single chapter / mp3 audio books
- [x] Unit tests passed.

Closes #242

### CODE REVIEW

File: BookPlayer/Player/PlayerViewController.swift [hunk 1]
```diff
@@ -177,7 +177,13 @@ class PlayerViewController: UIViewController, UIGestureRecognizerDelegate {
             items.append(self.sleepLabel)
         }
 
-        self.chaptersButton.isEnabled = self.currentBook.hasChapters
+        if self.currentBook.hasChapters {
```
Comment (GianniCarlo) [inline BookPlayer/Player/PlayerViewController.swift:180]: With this change, books that don't have chapters, have the chapters' button enabled, which shouldn't happen

Comment (mmurali88) [reply BookPlayer/Player/PlayerViewController.swift:180]: Ah my bad. Tested with one chapter book before pushing but missed item without any chapter. I have updated the button's default state to disabled in the story board so this works fine. LMK if this works.

Comment (GianniCarlo) [reply BookPlayer/Player/PlayerViewController.swift:180]: There's still one case not covered :/. If the user has autoplay in his library, or if it has in a playlist, a book with chapters, which is followed by a book with no chapters, when the player screen is reloaded for the next book (the one without chapters), the chapters button stays enabled

Comment (mmurali88) [reply BookPlayer/Player/PlayerViewController.swift]: Updated the crash on the corner case you mentioned. Tested the following scenarios

Autoplay order :
1. Book with chapters > 1  -> No chapter - > 1 chapter (Library/playlist) -> 2 Chapter
2. No chapter - >Book with chapters > 1 ->  1 chapter (Library/playlist) -> 2 Chapter 
3. Library No chapter item -> PLaylist(Book with chapters > 1) -> Library(1 chapter item)  -> Library(2 Chapter)

All of these have been tested with and without autoplay. Works as expected.

File: BookPlayer/Player/PlayerViewController.swift [hunk 2]
```diff
@@ -177,7 +177,11 @@ class PlayerViewController: UIViewController, UIGestureRecognizerDelegate {
             items.append(self.sleepLabel)
         }
 
-        self.chaptersButton.isEnabled = self.currentBook.hasChapters
+        guard let chapterCount = self.currentBook.chapters?.count else {
```
Comment (GianniCarlo) [inline BookPlayer/Player/PlayerViewController.swift:180]: This `guard` here if the `count` ever resolves to `nil` then the button `avRoutePickerBarButtonItem` after this block would never be added. Testing it, I found out that for books w/o chapters, it resolves to `0`, but I think that it's better if we change this guard and make it more explicit by replacing it for:
```swift
let chapterCount = self.currentBook.chapters?.count ?? 0
```

**Pull request merged**